src/modules/usart.rs

```diff
@@ -0,0 +1,258 @@
+    pub fn new(config: Config, baudrate: u32) -> Usart {
+        Usart {
+            config: config,
+            baudrate: baudrate,
+            usart: match config {
+                Config::Usart0(_) => usart::Usart::usart0(),
+                Config::Usart1(_) => usart::Usart::usart1(),
+                Config::Usart2(_) => usart::Usart::usart2(),
+            },
+            line_break: "\0\r",
+        }
+    }
+
+    /// Called in order to initialize the Usart according to it's config
+    pub fn init(&mut self) {
+        // Enable clock for HF peripherals
+        cmu::clock_select_set(cmu::Clock::HF, cmu::Select::HFRCO);
+
+        // Configure usart
+        let (usart_clock, usart) = match self.config {
+            Config::Usart0(_) => (cmu::Clock::USART0, usart::Usart::usart0()),
+            Config::Usart1(_) => (cmu::Clock::USART1, usart::Usart::usart1()),
+            Config::Usart2(_) => (cmu::Clock::USART2, usart::Usart::usart2()),
+        };
+        self.usart = usart;
+        cmu::clock_enable(usart_clock, true);
+        self.usart.init_async(&usart::InitAsync {
+            enable: usart::Enable::Disable,
+            baudrate: self.baudrate,
+            .. Default::default()
+        });
+
+        // Configure TX, RX, CLK, CS pins and Usart location
+        self.init_location_and_gpio();
+
+        // Enable Usart
+        self.usart.enable(usart::Enable::Enable);
+    }
+
+    /// Performs a blocking send of one `char`.
+    pub fn putc(&self, data: u8) {
+        self.usart.tx(data);
+    }
+
+    /// Blocks until one `char` is received.
+    pub fn getc(&self) -> u8 {
+        self.usart.rx()
+    }
+
+    /// Performs a blocking send of a `&str`
+    pub fn write_str(&self, string: &str) {
+        for c in string.chars() {
+            self.putc(c as u8);
+        }
+    }
+
+    /// Blocks and returns a `String` that is terminated by either one of the
+    /// characters in `line_break`.
+    ///
+    /// #Panics
+    /// Panics if it fails to parse the received characters to a UTF8 String.
+    pub fn read_string(&self) -> String {
+        let mut bytes = Vec::new();
+
+        loop {
+            match self.getc() as char {
+                c if self.line_break.contains_char(c) => { break; }
+                c => { bytes.push(c as u8); }
+            }
+        }
+
+        String::from_utf8(bytes).unwrap_or_else(|e| {
+            panic!("{}", e)
+        })
+    }
+
+    /// Returns `true` if a `char` is available to read.
+    pub fn readable(&self) -> bool {
+        (self.usart.IF & usart::STATUS_RXDATAV) != 0
+    }
+
+    /// Returns `true` if there is space available to write a `char`.
+    pub fn writeable(&self) -> bool {
+        (self.usart.IF & usart::STATUS_TXBL) != 0
+    }
+
+
+    fn init_location_and_gpio(&mut self) {
+        let location = match self.config {
+            Config::Usart0(loc) => {
+                match loc {
```

Comment:
I can, but I really don't think it would look any nicer. 
I  just realised that this code is pretty similar to one of the em_spi examples, instantiating the spi in a similar way.